core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1828,7 +1832,8 @@ class KafkaApis(val requestChannel: RequestChannel,
               errors.foreachEntry { (topicIdPartition, partitionResponse) =>
                 addResultAndMaybeComplete(topicIdPartition.topicPartition(), partitionResponse.error)
               }
-            }
+            },
+            transactionVersion = markerTransactionVersion
```

Comment:
What about`groupCoordinator.completeTransaction`?  It also need to check the epoch properly.